File: test-fe/src/service/ProductService.js
import axios from "axios";

const PRODUCT_API_BASE_URL='http://localhost:8386/product';
// const PRODUCT_DETAIL_API_URL=`http://localhost:8386/product/${id}`

class ProductService{
    async getProduct(){
        return await axios.get(PRODUCT_API_BASE_URL);
    };

    async getProductById(id) {
        return await axios.get(`http://localhost:8386/product/${id}`);
    }
}


export default new ProductService();

File: demo-be/src/main/java/com/example/demo/service/ProductService.java
package com.example.demo.service;

import com.example.demo.entity.Product;
import com.example.demo.model.request.AddProductRequest;
import com.example.demo.model.request.ProductRequest;
import com.example.demo.model.request.UpdateProductRequest;
import org.springframework.data.domain.Pageable;
import org.springframework.http.ResponseEntity;

public interface ProductService {
    ResponseEntity<?> getProduct(Pageable pageable);

    ResponseEntity<?> addProduct(AddProductRequest request);

    ResponseEntity<?> updateProduct(Long id, UpdateProductRequest request);

    ResponseEntity<?> deleteProduct(Long id);

    ResponseEntity<?> getProductById(Long id);
}
